chainerx_cc/chainerx/routines/indexing.cc

```diff
@@ -156,34 +156,33 @@ Array AddAt(const Array& a, const Array& indices, int8_t axis, const Array& b) {
 }  // namespace
 
 Array Take(const Array& a, const Array& indices, int8_t axis) {
-    // TODO(niboshi): Support other dtypes by casting
-    if (indices.dtype() != Dtype::kInt64) {
-        throw DtypeError(
-                std::string{"Only "} + GetDtypeName(Dtype::kInt64) + " is supported as indices, but given " +
-                GetDtypeName(indices.dtype()));
+    DtypeKind indices_kind = GetKind(indices.dtype());
+    if (!(indices_kind == DtypeKind::kInt || indices_kind == DtypeKind::kUInt)) {
+        throw DtypeError{"Dtype ", GetDtypeName(indices.dtype()), " cannot be used as an indices array."};
     }
+    const Array& indices_cast = indices.dtype() == Dtype::kInt64 ? indices : indices.AsType(Dtype::kInt64);
```

Comment:
Should we let the device implementation decide whether this cast is needed or not?

Reply:
Fixed as suggested.